File: Dataset/Leetcode/train/48/166.cpp
class Solution {
public:
    void XXX(vector<vector<int>>& matrix) {
        int temp=0,n=matrix.size()-1;
        int times = matrix.size()/2;//需要旋转的圈数2、3层只转最外边一层，4、5转最外边两圈；
        int cur_times = 0;
        while(cur_times<times)
        {
            int t = matrix.size()-cur_times-1;
            for(int i =cur_times;i<t;i++)
            {

                temp=matrix[cur_times][i];
                matrix[cur_times][i]=matrix[matrix.size()-i-1][cur_times];
                matrix[matrix.size()-i-1][cur_times]=matrix[matrix.size()-cur_times-1][matrix.size()-i-1];
                matrix[matrix.size()-cur_times-1][matrix.size()-i-1]=matrix[i][matrix.size()-cur_times-1];
                matrix[i][matrix.size()-cur_times-1]=temp;
            }
            cur_times++;
        } 
    }
};

File: Dataset/Leetcode/train/69/166.java
class Solution {
    public int XXX(int x) {
        if(x<2) return x;
        int l=1,r=x,m=1;
        while(l<=r){
            m=l+(r-l)/2;
            if(m>x/m) r=m-1;//先截取上界
            else if((m+1)>x/(m+1)) return m;//如果m^2<x，但是m+1^2＞x,则平方根就定下来了
            else {
                l=m+1;
            }
        }
        return l;
    }
}

